docs/core-concepts.md

```diff
@@ -97,22 +93,37 @@ const page = await browser.newPage();
 
 ## Execution contexts
 
-The Playwright API can be used in a JavaScript file running in a Node.js
-environment. When a browser instance is created, Playwright creates a
-bi-directional communication channel between the Node.js script and the browser.
+Playwright scripts run in your Node.js environment. You page scripts run in the page environment. Those environments don't intersect, they are running in different virtual machines in different processes and potentially on different computers.
+
+IMAGE PLACEHOLDER
 
-With this channel in place, Playwright can also execute JavaScript code in the
-context of the web page. This enables another execution context for automation
-scripts.
+Playwright exposes APIs to execute the JavaScript code in the context of the web page and allows calling back into the Node.js environment from your web page.
 
 ### Using the Evaluate API
 
 The [`page.evaluate`](https://github.com/microsoft/playwright/blob/master/docs/api.md#pageevaluatepagefunction-arg) API can run a JavaScript function in the context
 of the web page and bring results back to the Node.js environment. Globals like
-`window` and `document` can be used in `evaluate`.
+`window` and `document` along with the web page runtime can be used in `evaluate`.
+
+Right:
 
 ```js
+const data = { text: 'some data', value: 1 };
+// Pass |data| as a parameter.
+const result = await page.evaluate(data => {
```

Comment:
Do you think defining the function before the evaluate statement will make it easier to read?

Reply:
I don't think they will do it, basically they all use it in line